metadata_backend/api/handlers.py

```diff
@@ -1133,23 +1134,72 @@ async def _extract_xml_upload(req: Request, extract_one: bool = False) -> List[T
         part = cast(BodyPartReader, part)
         if not part:
             break
-        if extract_one and files:
+        if extract_one and (xml_files or csv_files):
             reason = "Only one file can be sent to this endpoint at a time."
             LOG.error(reason)
             raise web.HTTPBadRequest(reason=reason)
-        if part.name:
-            schema_type = part.name.lower()
-            if schema_type not in schema_types:
-                reason = f"Specified schema {schema_type} was not found."
-                LOG.error(reason)
-                raise web.HTTPNotFound(reason=reason)
-            data = []
-            while True:
-                chunk = await part.read_chunk()
-                if not chunk:
-                    break
-                data.append(chunk)
-            xml_content = "".join(x.decode("UTF-8") for x in data)
-            files.append((xml_content, schema_type))
-            LOG.debug(f"processed file in {schema_type}")
-    return sorted(files, key=lambda x: schema_types[x[1]]["priority"])
+
+        content, schema_type = await _extract_upload(part)
+
+        if is_xml:
```

Comment:
i think we should split this into its own function, or at least split `_multipart_content` into checking file type and returning file/content

Reply:
Good point. I have separated the content type check to another helper function